src/pudl/extract/eiaaeo.py

```diff
@@ -0,0 +1,252 @@
+        self, series_id: str, potential_parents: set[int]
+    ) -> pd.DataFrame:
+        # Use the graph info to figure out what case + category name this
+        # series belongs to
+
+        # note that case-names is a list, because we would be surprised if
+        # there were multiple case nodes with the same name reported in
+        # nx.ancestors
+        case_names = [
+            self.graph.nodes[a_id]["name"]
+            for a_id in nx.ancestors(self.graph, series_id)
+            if a_id in self.__cases
+        ]
+        if len(case_names) != 1:
+            raise ValueError(
+                f"Found multiple AEO cases for series {series_id}: {case_names}"
+            )
+        case_name = case_names[0]
+
+        # whereas parent_names is a set, because we expect many of the
+        # predecessors to share a name
+        parent_names = {
+            self.graph.nodes[p_id]["name"]
+            for p_id in self.graph.predecessors(series_id)
+            if p_id in potential_parents
+        }
+        if len(parent_names) != 1:
+            raise ValueError(
+                f"Found multiple parents for series {series_id}: {parent_names}"
+            )
+        parent_name = parent_names.pop()
+
+        # in addition to case_name and parent_name, we get some information
+        # from the actual series itself.
+        series = self.graph.nodes[series_id]
+        records = (
+            {
+                "date": d[0],
+                "value": d[1],
+                "units": series["units"],
+                "series_name": series["name"],
+                "category_name": parent_name,
+                "case": case_name,
+            }
+            for d in series["data"]
+        )
+        return records
+
+    def get_table(self, table_number: int) -> pd.DataFrame:
+        """Get a specific table number as a DataFrame."""
+        matching_category_ids = {
+            n_id
+            for n_id in self.graph
+            if self.__sanitize(self.graph.nodes.get(n_id).get("name", "")).startswith(
+                f"table_{table_number}_"
+            )
+        }
+
+        # many series belong to more than one category, hence turning this into a set
+        matching_series = set(
+            itertools.chain.from_iterable(
+                self.graph.nodes[n_id].get("childseries", [])
+                for n_id in matching_category_ids
+            )
+        )
+
+        series_records = itertools.chain.from_iterable(
+            self.__series_to_records(series_id, potential_parents=matching_category_ids)
+            for series_id in matching_series
+        )
+        return AEOTable(pd.DataFrame.from_records(series_records))
+
+
+@multi_asset(
+    outs={
+        "raw_eiaaeo__electric_power_projections_regional": AssetOut(is_required=False),
+        "raw_eiaaeo__renewable_energy_regional": AssetOut(is_required=False),
+    },
+    can_subset=True,
+    required_resource_keys={"datastore"},
```

Comment:
Do we want to add ETL settings to the required resource keys and define classes for describing them in `pudl.settings` to allow us to control what data is being extracted, rather than hard-coding 2023?